src/Native/configure.cmake

```diff
@@ -196,13 +199,53 @@ check_function_exists(
     kqueue
     HAVE_KQUEUE)
 
+check_cxx_source_compiles(
+     "
+     #include <sys/types.h>
+     #include <netdb.h>
+
+     int main()
+     {
+         const void* addr;
+         socklen_t len;
+         int type;
+         struct hostent* result;
+         char* buffer;
+         size_t buflen;
+         struct hostent** entry;
+         int* error;
+         gethostbyaddr_r(addr,  len, type, result, buffer, buflen, entry, error);
+         return 0;
+     }
+     "
+     HAVE_GETHOSTBYADDR_R)
+
+check_cxx_source_compiles(
+     "
+     #include <sys/types.h>
+     #include <netdb.h>
+
+     int main()
+     {
+         const char* hostname;
+         struct hostent* result;
+         char* buffer;
+         size_t buflen;
+         struct hostent** entry;
+         int* error;
+         gethostbyname_r(hostname, result, buffer, buflen, entry, error);
+         return 0;
+     }
+     "
+     HAVE_GETHOSTBYNAME_R)
+
 check_function_exists(
-    gethostbyname_r
-    HAVE_GETHOSTBYNAME_R)
+    getaddrinfo
+    HAVE_GETADDRINFO)
 
 check_function_exists(
-    gethostbyaddr_r
-    HAVE_GETHOSTBYADDR_R)
+    getnameinfo
+    HAVE_GETNAMEINFO)
```

Comment:
That's correct. Thank you for making it crystal clear. :+1: 
I have fixed it in 8c4bb2a and tested on both FreeBSD and NetBSD locally.
The only thing I haven't reverted is the new way of detecting existing HAVE_GETHOSTBYADDR_R, because cmake search it the function in all system headers, although those are not available in netdb.h headers on netbsd, it is picking up from X11 header, where those are defined (hence false-positive).